src/core/evaluator.js

```diff
@@ -1676,13 +1689,19 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
               }
               break;
           } // switch
+          if (textContent.items.length >= sink.desiredSize) {
+            // wait for ready, if we reach highWaterMark.
+            stop = true;
+            break;
+          }
         } // while
         if (stop) {
           next(deferred);
           return;
         }
         flushTextContentItem();
-        resolve(textContent);
+        enqueueChunk();
+        resolve();
```

Comment:
> resolve() actually resolves the outer promise -- just removing it will block its resolution

I'm not sure that I follow here.
It might not have been clear from my comment, since I couldn't add it on the line in question, but I'd assume that we *also* need to update line https://github.com/mozilla/pdf.js/blob/593dec1bb7aec1802abf8268137b0f7adab2ae32/src/core/evaluator.js#L1692.